benchmark.go

```diff
@@ -126,15 +127,30 @@ func runBenchmark(out output, logger *zap.Logger, allOpts Options, m benchmarkMe
 		out.Fatalf("Failed to warmup connections for benchmark: %v", err)
 	}
 
-	statter, err := statsd.NewClient(logger, opts.StatsdHostPort, allOpts.TOpts.ServiceName, allOpts.ROpts.Procedure)
+	globalStatter, err := statsd.NewClient(logger, opts.StatsdHostPort, allOpts.TOpts.ServiceName, allOpts.ROpts.Procedure)
 	if err != nil {
 		out.Fatalf("Failed to create statsd client for benchmark: %v", err)
 	}
 
 	var wg sync.WaitGroup
 	states := make([]*benchmarkState, len(connections)*opts.Concurrency)
-	for i := range states {
-		states[i] = newBenchmarkState(statter)
+
+	for i, c := range connections {
+		statter := globalStatter
+
+		if opts.PerPeerStats {
+			// If per-peer stats are enabled, dual emit metrics to the original value
+			// and the per-peer value.
+			prefix := fmt.Sprintf("peer.%v.", c.peerID)
+			statter = statsd.MultiClient{
+				globalStatter,
+				statsd.NewPrefixedClient(globalStatter, prefix),
```

Comment:
updated to use statter in both places (`MultiClient` and `PrefixedClient`) since it seemed inconsistent. The line looks a little odd now though, since we have 3 references to `statter`